feat(Grammar): drop '#' for comments

adnrs96 opened the pull request:
- We are doing this because '#' will be used for Data feature.
- We add the new syntax for comments i.e.
  /*
	long comment
  */
  // short comment
  a = 1 // in-line comment

Fixes: #1346

wilzbach commented on tests/e2e/multiline_comments4.story:
```diff
@@ -1,15 +1,16 @@
 .
-###b = 0
+*/
+b = 0
```
```suggestion
```

The previous code wasn't supposed to execute `b = 0`.

adnrs96 replied:
I will address this by removing `b = 0` as you suggested.

I just kept it around since it wasn't really causing any issues I guess and I couldn't allow
```
/* 
something
*/ b = 0
```

type of syntax to exist. If I were to do that it will be an error

pep8speaks commented:
Hello @adnrs96! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`storyscript/parser/Grammar.py`](https://github.com/storyscript/storyscript/blob/96c6dc383ce5bf45408e58587f8a4c981687ef52/storyscript/parser/Grammar.py):

> [Line 338:80](https://github.com/storyscript/storyscript/blob/96c6dc383ce5bf45408e58587f8a4c981687ef52/storyscript/parser/Grammar.py#L338): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)

adnrs96 commented:
@wilzbach I updated this PR and added some modifications to the grammar in a separate commit which I believe to improve the comments parsing. Read commit message for details.
Thanks!
